instrumentation/net/http/otelhttp/transport.go

```diff
@@ -263,7 +264,16 @@ func (wb *wrappedBody) Read(b []byte) (int, error) {
 	return n, err
 }
 
+// recordBytesRead is a function that ensures the number of bytes read is recorded once and only once.
+func (wb *wrappedBody) recordBytesRead() {
+	if wb.recorded.CompareAndSwap(false, true) {
```

Comment:
Maybe add a comment here that we do not need to use `sync.Once` as it is OK that have "eventual consistency"? Basically if there are two goroutines running in parallel that call `recordBytesRead`, one can return before the second called `wb.record(wb.read.Load())`. It is OK and using `CompareAndSwap` instead of `sync.Once` is a performance optimization, but I think it is worth adding a comment.

Reply:
Interesting, I'm relatively new to go so I had not known about `sync.Once`. I just wanted semantics close to Java's atomic boolean so that is why I chose `atomic.bool`. Anyway, added comments in 38861442146dafb02345a3cd06d6bd96030c77f6